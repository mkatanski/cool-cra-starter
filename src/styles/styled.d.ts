/* eslint-disable @typescript-eslint/no-empty-interface */
import { DefaultTheme } from "styled-components";
import { StyledTheme } from "styles/_types";
import { palette } from "styles/palette";

declare module "styled-components" {
  export interface DefaultTheme {
    elements: StyledTheme;
    palette;
  }
}
