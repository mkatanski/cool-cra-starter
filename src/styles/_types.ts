export type InteractiveElement = {
  default: string;
  hover?: string;
};

export type TextElement = {
  primary: string;
  secondary?: string;
  inverse?: string;
};

export type InputElement = {
  background: InteractiveElement;
  text: TextElement;
  focus?: string;
};

export type TableElement = {
  border: string;
  background: string;
  text: TextElement;
};

export type StyledTheme = {
  container: {
    background: string;
    text: TextElement;
  };

  button: {
    background: InteractiveElement;
    text: InteractiveElement;
  };

  input: InputElement;

  table: TableElement;
  tableHeader: TableElement;
  tableRowEven: TableElement;
  tableRowOdd: TableElement;
};
