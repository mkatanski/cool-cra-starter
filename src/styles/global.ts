import { createGlobalStyle } from "styled-components";

export const MainStyle = createGlobalStyle`
  *, *::before, *::after {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }
  body {
    margin: 0;
    padding: 0;
    font-family: 'Roboto', Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    background: #f5f5f5;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    flex: 1;
    background: ${({ theme }) => theme.elements.container.background};
    color: ${({ theme }) => theme.elements.container.text.primary};
    scroll-behavior: smooth;
  }
  code {
    font-family: Consolas, 'Courier New', monospace;
  }
`;
