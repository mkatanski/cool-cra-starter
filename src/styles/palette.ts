export const palette = {
  concrete: "#f2f2f2",
  bahamaBlue: "#03658C",
  azalea: "#F2BDD6",
  charm: "#D984AC",
  cranberry: "#D95995",

  holly: "#012615",
  outerSpace: "#284035",
  wewak: "#F2A7A0",
  opium: "#8C6B68",
  beautyBush: "#F2C4C4",
};
