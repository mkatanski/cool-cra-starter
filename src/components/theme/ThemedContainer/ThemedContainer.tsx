import React, { useContext } from "react";
import { ThemeProvider } from "styled-components";
import { StyledTheme } from "styles/_types";
import { palette } from "styles/palette";
import { ThemeName } from "config/themes";
import { ThemeContext } from "components/theme/ThemeProvider";

export type ThemedContainerProps = {
  themes: { [key in ThemeName]: StyledTheme };
  children: React.ReactNode;
};

export const ThemedContainer: React.FC<ThemedContainerProps> = ({
  children,
  themes,
}: ThemedContainerProps) => {
  const { themeMode } = useContext(ThemeContext);
  const theme = themes[themeMode];

  return (
    <ThemeProvider
      theme={{
        elements: theme,
        palette: palette,
      }}
    >
      {children}
    </ThemeProvider>
  );
};
