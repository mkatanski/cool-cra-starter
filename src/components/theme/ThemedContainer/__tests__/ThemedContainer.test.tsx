import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

/* Uncomment below if needed */
// import { withThemeProvider, withDesktopSize } from 'testsHelpers';
// import { render, getByTestId } from "@testing-library/react";

import { ThemedContainer } from "../ThemedContainer";

/* Uncomment below if You need/want to use enzyme */
// import enzyme from "enzyme";

/* Component mock example */
// jest.mock("[module]", () => ({
//   [ComponentName]: ({children, ...rest}: React.PropsWithChildren<{}>) => (
//     <div id="[ComponentName]" {...rest}>{children}</div>
//   )
// }));

import { StyledTheme } from "styles/_types";

const TestTheme: StyledTheme = {
  container: {
    background: "container.background",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  button: {
    text: {
      default: "palette.holly",
      hover: "palette.outerSpace",
    },
    background: {
      default: "palette.azalea",
      hover: "palette.charm",
    },
  },

  input: {
    background: {
      default: "palette.azalea",
      hover: "palette.azalea",
    },
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  table: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableHeader: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowEven: {
    background: "palette.wewak",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },

  tableRowOdd: {
    background: "palette.beautyBush",
    border: "palette.wewak",
    text: {
      primary: "palette.holly",
      inverse: "palette.beautyBush",
      secondary: "palette.outerSpace",
    },
  },
};

describe("components", () => {
  describe("ThemedContainer", () => {
    it("should render ThemedContainer component", () => {
      const tree = TestRenderer.create(
        <ThemedContainer themes={{ dark: TestTheme, light: TestTheme }}>
          Test
        </ThemedContainer>
      );

      expect(tree).toMatchSnapshot();
    });
  });
});
