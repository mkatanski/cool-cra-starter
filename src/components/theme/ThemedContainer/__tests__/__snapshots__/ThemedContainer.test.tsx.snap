// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ThemedContainer should render ThemedContainer component 1`] = `"Test"`;
